### PR TITLE
Raghav1783 fix issue 2719

### LINKED ISSUE #2719: [BUG] balance shows vertically when scrolling down.
### Please confirm the following

- [X] I checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate problems

### Describe the bug

The balance on the home screen gets compressed and appears vertical when scrolling down. It happens both if the balance is hidden and if it is shown.

Maybe it is because I have custom starting date thus longer month name.

### To Reproduce

1. Open the app
2. Scroll down 

### Expected behavior

The balance (or stars) should be displayed horizontally on one line. Maybe pushed down on the next line if it doesn't fit.

### Screenshots

Screenshot_20230928_103826_Ivy Wallet

### Smartphone

Samsung Galaxy S20 FE

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Raghav1783**
## Pull Request (PR) Checklist
Please check if your pull request fulfills the following requirements:
- [x] The PR is submitted to the `main` branch.
- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the "Contributing Rules".
- [x] I've read the [Architecture Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Architecture.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [x] 🎬 I've attached a **screen recoding** of the changes. 

> Tip: drag & drop the video to the PR description.

https://github.com/Ivy-Apps/ivy-wallet/assets/134570234/5f3d647e-98d0-4463-b983-39361baae331

## What's changed?

before | 
after

Describe with a few bullets **what's new:**
- a fix the balance overflowing bug.
- b when there is not enough space balance and currency will be displayed in 2 lines
- c
- d

> 💡 Tip: Please, attach screenshots and screen recordings. It helps a lot!

## Risk Factors

**What may go wrong if we merge your PR?**

- a nothing
- b
- c

**In what cases your code won't work?**

- a It works fine
- b
- c

## Does this PR closes any GitHub Issues?

Check **[Ivy Wallet Issues](https://github.com/Ivy-Apps/ivy-wallet/issues)**.

- Closes #2719

> Replace `ISSUE_NUMBER` with your issue number (for example Closes #1234). If you've done that correctly, you'll see the issue title linked when previewing your PR description.

## Troubleshooting CI failures

If you see any of the PR checks failing (❌) go to [Actions](https://github.com/Ivy-Apps/ivy-wallet/actions) and find it there. Or simply click "Details" next to the failed check and explore the logs to see why it has failed.

### Detekt
[Detekt](https://detekt.dev/) is a static code analyzer for Kotlin that we use to enforce code readibility and good practices.

**To run Detekt locally:**
```
./gradlew detekt
```

If the Detekt errors are caused by a legacy code, you can suppress them using a basline.

**Detekt baseline** (not recommended)
```
./scripts/detektBaseline.sh
```

### Lint

We use the [standard Android Lint](https://developer.android.com/studio/write/lint) plus [Slack's compose-lints](https://slackhq.github.io/compose-lints/) as an addition to enforce proper Compose usage.

**To run Lint locally:**
```
./scripts/lint.sh
```

If the Lint errors are caused by a legacy code, you can suppress them using a basline.

**Lint baseline** (not recommended)
```
./scripts/lintBaseline.sh
```

### Unit tests

If this job is failing this means that your changes break an existing unit test. You must identify the failing tests and fix your code.

**To run the Unit tests locally:**
```
./gradlew testDebugUnitTest
```

### CODE REVIEW

File: screen-home/src/main/java/com/ivy/home/HomeHeader.kt [hunk 1]
```diff
@@ -157,6 +157,17 @@ private fun HeaderStickyRow(
 
             // Balance mini row
             if (percentExpanded < 1f) {
+                val overflowLengthOfBalance = 7
+                val overflowLengthOfMonthRange = 12
+                val lengthOfCurrencyAndBalance = (currency + balance.toString()).length
```
Comment (ILIYANGERMANOV) [inline screen-home/src/main/java/com/ivy/home/HomeHeader.kt:162]: Use `remember` + `deriveStateOf` to make it more optimal.
```kotlin
val overflow by remember(... things that depend ...) {
   deriveStateOf {
        do calculations here
    }
}
```

**Review by ILIYANGERMANOV — COMMENTED** [review]
Looks good! We just need to optimize it using `remember` and `deriveStateOf`. Also you need to fix the CI, use the info provided in the PR description.

File: screen-home/src/main/java/com/ivy/home/HomeHeader.kt (hunk 1 above)
Comment (Raghav1783) [reply screen-home/src/main/java/com/ivy/home/HomeHeader.kt:162]: i am not able to resolve these lint issues

Comment (ILIYANGERMANOV) [reply screen-home/src/main/java/com/ivy/home/HomeHeader.kt:164]: @Raghav1783 Detekt is pretty easy. Just extra magic numbers as constants and remove unused imports.
```
home/runner/work/ivy-wallet/ivy-wallet/screen-budgets/src/main/java/com/ivy/budgets/BudgetModal.kt:1:1: Unused import [NoUnusedImports]
/home/runner/work/ivy-wallet/ivy-wallet/screen-budgets/src/main/java/com/ivy/budgets/BudgetModal.kt:1:1: Unused import [NoUnusedImports]
/home/runner/work/ivy-wallet/ivy-wallet/screen-home/src/main/java/com/ivy/home/HomeHeader.kt:162:47: This expression contains a magic number. Consider defining it to a well named constant. [MagicNumber]
/home/runner/work/ivy-wallet/ivy-wallet/screen-home/src/main/java/com/ivy/home/HomeHeader.kt:163:50: This expression contains a magic number. Consider defining it to a well named constant. [MagicNumber]
```

Lint, I don't know screenshot me the Lint report. If it's hard to resolve you add a baseline to suppress them.

**Comment by ILIYANGERMANOV** [comment]
@Raghav1783 lint seems also easy just reorder the parameters of
```
[../../../temp-legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/components/BalanceRow.kt](content://media/temp-legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/components/BalanceRow.kt):60: Parameters in a composable function should be ordered following this pattern: params without defaults, modifiers, params with defaults and optionally, a trailing function that might not have a default param.
Current params are: [modifier: Modifier = Modifier, textColor: Color = UI.colors.pureInverse, currency: String, balance: Double, balanceAmountPrefix: String? = null, currencyUpfront: Boolean = true, shortenBigNumbers: Boolean = false, hiddenMode: Boolean = false, overflow: Boolean = false] but should be [currency: String, balance: Double, modifier: Modifier = Modifier, textColor: Color = UI.colors.pureInverse, balanceAmountPrefix: String? = null, currencyUpfront: Boolean = true, shortenBigNumbers: Boolean = false, hiddenMode: Boolean = false, overflow: Boolean = false].
See https://slackhq.github.io/compose-lints/rules/#ordering-composable-parameters-properly for more information.
```

**Parameters in a composable function should be ordered following this pattern: params without defaults, modifiers, params with defaults and optionally, a trailing function that might not have a default param.**

**Comment by Raghav1783** [comment]
how to see Lint report

**Comment by ILIYANGERMANOV** [comment]
> how to see Lint report

@Raghav1783 it's attached at the bottom 
https://github.com/Ivy-Apps/ivy-wallet/actions/runs/6441777239

**Comment by Raghav1783** [comment]
this looks fine ig

**Comment by ILIYANGERMANOV** [comment]
> this looks fine ig

you fixed Lint, what's left is Detekt:
```
./gradlew detekt
```

Run this command to see the error

**Comment by Raghav1783** [comment]
hey @ILIYANGERMANOV i checked detekt error, 1 error is related to OptionalWhenBraces is deprecated which i saw is another issue that is assigned to someone else, and the other issue is the magic number one, i wanted to ask in which file should i define the constant variable.

**Comment by ILIYANGERMANOV** [comment]
> hey @ILIYANGERMANOV i checked detekt error, 1 error is related to OptionalWhenBraces is deprecated which i saw is another issue that is assigned to someone else, and the other issue is the magic number one, i wanted to ask in which file should i define the constant variable.

Yes, the only issue that needs to be addressed is the MagicNumbers. @Raghav1783 just extra them as a constants in the same file. Preferably `private const val` at the top of the file

File: screen-home/src/main/java/com/ivy/home/HomeHeader.kt [hunk 2]
```diff
@@ -57,6 +59,8 @@ import com.ivy.wallet.ui.theme.components.IvyIcon
 import com.ivy.wallet.ui.theme.components.IvyOutlinedButton
 import com.ivy.wallet.ui.theme.wallet.AmountCurrencyB1
 import kotlin.math.absoluteValue
+private const val overflowLengthOfBalance = 7
```
Comment (ILIYANGERMANOV) [inline screen-home/src/main/java/com/ivy/home/HomeHeader.kt:62]: - Add a blank line above
- OVERFLOW_LENGTH_OF_BALANCE

**Comment by ILIYANGERMANOV** [comment]
That's bug in Detekt, use:
```
./scripts/detektBaseline.sh
```

to add a baseline. The bug is that we don't allow "_" for some reason

**Comment by ILIYANGERMANOV** [comment]
Or just name it `OverflowLengthOfBalance` so Detekt don't complain

**Comment by Raghav1783** [comment]
> Or just name it `OverflowLengthOfBalance` so Detekt don't complain
I was about do the same when you commented use OVERFLOW_LENGTH_OF_BALANCE

**Review by ILIYANGERMANOV — APPROVED** [review]

**Pull request merged**